chia/data_layer/data_layer.py

```diff
@@ -102,33 +108,24 @@ async def batch_update(
         changelist: List[Dict[str, Any]],
         fee: uint64,
     ) -> TransactionRecord:
-        hint_keys_values = await self.data_store.get_keys_values_dict(tree_id)
-        for change in changelist:
-            if change["action"] == "insert":
-                key = change["key"]
-                value = change["value"]
-                reference_node_hash = change.get("reference_node_hash")
-                side = change.get("side")
-                if reference_node_hash or side:
-                    await self.data_store.insert(key, value, tree_id, reference_node_hash, side, hint_keys_values)
-                await self.data_store.autoinsert(key, value, tree_id, hint_keys_values)
+        old_root = await self.data_store.get_tree_root(tree_id=tree_id)
+        rollback_generation = 0 if old_root is None else old_root.generation
+        try:
+            await self.data_store.insert_batch(tree_id, changelist)
+            root = await self.data_store.get_tree_root(tree_id=tree_id)
+            # todo return empty node hash from get_tree_root
```

Comment:
this is old code (but it appears new due to changing tabs) - I think the point is for `data_store` to return `bytes32([0]*32)` instead of `None`, so we are consistent with wallet. I think Almog did some work here on this:
https://github.com/Chia-Network/chia-blockchain/pull/10475